### PR TITLE
add radio component

### PR DESCRIPTION
**Pull request by 1297947244**

### CODE REVIEW

File: docs/src/components/fields/RadioGroupField.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { IField, IFieldDef } from 'react-dynamic-formbuilder';
+import { RadioGroupTextField, RadioGroupTextBuilder } from './RadioGroupTextField';
+import KeyValueDisplay from './KeyValueDisplay';
+import RadioGroupOptionEditor from './RadioGroupOptionEditor';
+
+const Type = "Teamious.radioButton";
+const Field: IField = {
+    id: '',
+    label: '单选按钮',
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupField.ts:9]: ??

File: docs/src/components/fields/RadioGroupField.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ],
+    options: {
+        required: true
+    }
+};
+
+export const RadioGroupField: IFieldDef = {
+    type: Type,
+    field: Field,
+    displayName: '单选按钮',
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupField.ts:36]: ??

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { FormControl, Checkbox } from 'react-bootstrap';
+import * as assign from 'object-assign';
+
+import { IFieldOptionEditorProps } from 'react-dynamic-formbuilder';
+
+interface IState { }
+
+
+export default class RadioGroupOptionEditor extends React.PureComponent<IFieldOptionEditorProps, IState> {
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx:10]: React.PureComponent<IFieldOptionEditorProps, void>

if not state.

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    render() {
+        const { label, fields } = this.props.field;
+        const { required, unique } = this.props.field.options;
+        return (
+            <div>
+                <div>
+                    <span>Label</span>
+                    <FormControl type='text' value={label} onChange={this.onLabelChange} />
+                    <span>选项设置</span>
+                    {fields.map((item) => {
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx:33]: Maybe we have a common control already, check how the dropdown field is implemented.

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        this.props.onChange(field);
+    }
+
+    private addOption() {
+        //动态添加option
+        let field = assign({}, this.props.field);
+        let new_option = {
+            id: (this.addOptionId++).toString(),
+            label: "选项"+this.addOptionId,
+            type: "ABCDEFG"
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx:96]: what's this type used for?

File: docs/src/components/fields/RadioGroupField.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+const Type = "Teamious.radioButton";
+const Field: IField = {
+    id: '',
+    label: '单选按钮',
+    type: Type,
+    fields: [
+        {
+            id: '001',
+            label: 'JAVA',
+            type: 'ABCDEFG',
```
Comment (linweigao) [inline docs/src/components/fields/RadioGroupField.ts:15]: What's this type used for?

**Review by linweigao — CHANGES_REQUESTED** [review]

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                    <span>Label</span>
+                    <FormControl type='text' value={label} onChange={this.onLabelChange} />
+                    <span>选项设置</span>
+                    {fields.map((item) => {
+                        return <div className="dotted-box" key={item.id}>
+                            <a
+                                href="javascript:void(0)"
+                                className="box-close"
+                                onClick={this.removeOption.bind(this, item)}
+                            >X</a>
```
Comment (pleasureswx123) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx:39]: X是否可以用html特殊字符替代？

File: docs/src/components/fields/RadioGroupOptionEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+    private onRequiredChange(event: any) {
+        let field = assign({}, this.props.field);
+        field.options.required = !this.props.field.options.required;
+        this.props.onChange(field);
+    }
+
+    private onLabelValueChange(event: any) {
+        let field = assign({}, this.props.field);
+        field.fields.map((item, index) => {
```
Comment (pleasureswx123) [inline docs/src/components/fields/RadioGroupOptionEditor.tsx:72]: fields属性是否一定存在？

File: docs/src/components/fields/RadioGroupField.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+        },
+        {
+            id: '002',
+            label: 'WEB',
+            type: 'ABCDEFG',
+        },
+        {
+            id: '003',
+            label: 'IOS',
```
Comment (pleasureswx123) [inline docs/src/components/fields/RadioGroupField.ts:24]: value是否可以提出？作为常量？

**Pull request closed without merging**